cpp/arcticdb/version/test/test_sorting_info_state_machine.cpp

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <gtest/gtest.h>
+
+#include <bitmagic/bm.h>
+#include <bitmagic/bmserial.h>
+#include <arcticdb/util/buffer.hpp>
+#include <arcticdb/util/bitset.hpp>
+
+#include <msgpack.hpp>
+#include <arcticdb/util/test/generators.hpp>
+#include <arcticdb/version/version_store_api.hpp>
+
+
+TEST(Sorting, ExistingUnknown) {
+    using namespace arcticdb;
+    using namespace arcticdb::proto;
+
+    descriptors::SortedValue existing_frame = descriptors::SortedValue::UNKNOWN;
+
+    descriptors::SortedValue input_frame = descriptors::SortedValue::UNKNOWN;
+    ASSERT_EQ(deduce_sorted(existing_frame, input_frame), descriptors::SortedValue::UNKNOWN);
```

Comment:
It will be easier to follow if both parameters are in-line instead of variables, so the assertion line more closely resembles a table.

Actually, the traditional way of implementing a state machine is a look-up table which wouldn't require any tests.

Reply:
Consider this:

```cpp
    //               existing_frame, input_frame
    ASSERT_EQ(deduce_sorted(UNKNOWN, UNKNOWN   ), UNKNOWN);
    ASSERT_EQ(deduce_sorted(UNKNOWN, ASCENDING ), UNKNOWN);
    ASSERT_EQ(deduce_sorted(UNKNOWN, DESCENDING), UNKNOWN);
    ASSERT_EQ(deduce_sorted(UNKNOWN, UNSORTED  ), UNSORTED);
```

I would say that it's equally clear what the common argument is, but also much less visual noise.

Reply:
yes, I was thinking in exactly the same option, the only caviat is that it needs the comment bo be added...but fine

Reply:
done